Include groups claim steps to Okta SSO page (#74)
paralus/website: docs/Single Sign On/okta.md

In the dialog that opens, choose **OIDC - Open ID Connect** as the **Sign-in Method** and **Web Application** as the **Application Type.**

<img src="/img/docs/oidc-okta-1.png" alt="Creating New Okta OAuth App" height="75%" width="75%"/>

Leave the **Application Callback URL** empty for now and Register the application.

Based on who can access the Paralus app integration, select **Controlled access** options.

Note down the `Client ID` and `Client Secret` as those will be required to configure identity provider for Paralus.

### Add a custom claim for groups mapping

Paralus comes with the in-built functionality for mapping Identity

> provider (IdP) groups to Paralus. This is useful to manage different

Reviewer: `This is useful to manage different Paralus access to the IdP users through IdP groups they are part of. It eases the access-management process for Parlaus administrators`

We can rephrase this to "_This makes access management simpler. As an administrator, you don't need to manage access for an individual user but for a group of users_"

Paralus access to the IdP users through IdP groups they are part
of. It eases the access-management process for Parlaus

> administrators. See the limitation below.

Reviewer: `See limitation below` - can be rephrased to 

_This however has a limitation. To map the groups correctly, the group names in Okta and Paralus must be the same. This way, the IdP users belonging to a group can access Paralus resources based on the permissions assigned to the group in Paralus. For example, you create a group called `DevTeam` in Paralus and the same group in Okta as well. Assign permissions to the Paralus group and add users to the Okta group. Now, when an IdP user belonging to the `DevTeam` group logs into Paralus using Okta, they will inherit the permissions set for the group. Read how to [manage users and groups in Okta](https://help.okta.com/oie/en-us/Content/Topics/users-groups-profiles/usgp-groups-main.htm)_

In order to achieve successful groups mapping, both Okta group and
Paralus group must match their names. Once it matches, then IdP users
can access the Paralus resources based on permissions assigned to
matching Paralus group. Read how to manage Okta users and groups in
[Okta
documentation](https://help.okta.com/oie/en-us/Content/Topics/users-groups-profiles/usgp-groups-main.htm).

> Okta by default does not return groups information in OIdC

Reviewer: ```suggestion
Okta by default does not return groups information in OIDC
```

Also, can we link to the Okta document that mentions the scopes so the reader can see that groups are not returned? I found this link: https://auth0.com/docs/get-started/apis/scopes/openid-connect-scopes

Author: This is generic document about OIDC scopes and standard claims. I am not able to find a document from Okta.

authentication event. You need to add a custom claim in Okta
authorization server that provides groups data. There are the
necessary steps to add a claim:

1. Navigate to `Security -> API` and select the default authorization
   server. Click on **Add Claim**.
2. In the dialog that opens, set _Name_ to **groups**.
3. Choose _Include in token type_ as **ID Token** and **Always**.
4. Select _Value type_ to **Groups**.
5. Set the _Filter_ to **Match regex** and specify **.*** as a
   pattern.
6. Select _Include in_ to **Any scope**.

<img src="/img/docs/oidc-okta-2.png" alt="Added groups claim to Okta authorization server" height="75%" width="75%"/>

> > **Limitation:** Currently after the IdP user logged in to Paralus,

Reviewer: ```suggestion
> **Limitation:** Currently after the IdP user is logged in to Paralus,
```

> > any change in Okta groups does not reflect on Paralus. This is a

Reviewer: ```suggestion
> any change in Okta groups does not reflect in Paralus. This is a
```

> > known gap. It is being tracked

Reviewer: ```suggestion
> known gap. 
```

> > [here](https://github.com/paralus/paralus/issues/2).

Reviewer: ```suggestion
> This is being tracked as a part of this [issue](https://github.com/paralus/paralus/issues/2).
```

From the next screen copy the `Callback URL` and paste it in the callback URL for the Okta OAuth app created in the earlier step.

> On the **Mapper Configuration** screen, provide `https://raw.githubusercontent.com/paralus/paralus/main/_kratos/oidc-mappers/okta.jsonnet` as the mapper url. The mapper url has used [the custom claim added earlier for groups](#add-a-custom-claim-for-groups-mapping).

Reviewer: ```suggestion
On the **Mapper Configuration** screen, provide `https://raw.githubusercontent.com/paralus/paralus/main/_kratos/oidc-mappers/okta.jsonnet` as the mapper url. The mapper url uses [the custom claim added earlier for groups](#add-a-custom-claim-for-groups-mapping).
```
